src/lsdb/io/to_hats.py

```diff
@@ -176,34 +170,9 @@ def create_modified_catalog_structure(
     new_hc_structure.catalog_path = catalog_base_dir
     new_hc_structure.catalog_base_dir = hc.io.file_io.get_upath(catalog_base_dir)
     new_hc_structure.on_disk = True
-    new_hc_structure.catalog_info = dataclasses.replace(
-        new_hc_structure.catalog_info, catalog_name=catalog_name, **kwargs
-    )
-    return new_hc_structure
-
-
-def _get_provenance_info(catalog_structure: HCHealpixDataset) -> dict:
-    """Fill all known information in a dictionary for provenance tracking.
-
-    Args:
-        catalog_structure (HCHealpixDataset): The catalog structure
 
-    Returns:
-        dictionary with all argument_name -> argument_value as key -> value pairs.
-    """
-    structure_args = {
-        "catalog_name": catalog_structure.catalog_name,
-        "output_path": catalog_structure.catalog_path,
-        "output_catalog_name": catalog_structure.catalog_name,
-        "catalog_path": catalog_structure.catalog_path,
-    }
-    args = {
-        **structure_args,
-        **dataclasses.asdict(catalog_structure.catalog_info),
-    }
-    provenance_info = {
-        "tool_name": "lsdb",
-        "version": version("lsdb"),
-        "runtime_args": args,
-    }
-    return provenance_info
+    new_hc_structure.catalog_info = new_hc_structure.catalog_info.model_copy()
```

Comment:
```suggestion
    new_hc_structure.catalog_info = new_hc_structure.catalog_info.model_copy(update=dict(
        catalog_name=catalog_name,
        **kwargs
    ))
```
In this case, I do think the above pattern is worth preferring to the loop with `setattr`.

Reply:
Although, as before, `setattr` will do the spell check that `update=` doesn't.

Reply:
Dolphinately.

Reply:
> Dolphinately.

I was wrong about that: Pydantic models don't notice the misspellings *anyway*, unless `Config: extra = 'forbid'`, and it sounds like we don't do that.  (And even if we did, `update=` still wouldn't check.)